delivery_price_rule_volumetric_weight/models/delivery_carrier.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+from odoo.tools.safe_eval import safe_eval
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = "delivery.carrier"
+
+    def _get_price_available(self, order):
+        self.ensure_one()
+        self = self.sudo()
+        order = order.sudo()
+        total = weight = volumetric_weight = volume = quantity = 0
+        total_delivery = 0.0
+        for line in order.order_line.filtered(
+            lambda x: x.state != "cancel" and x.product_id
+        ):
+            if line.is_delivery:
+                total_delivery += line.price_total
+            if line.is_delivery:
+                continue
```

Comment:
No need to check twice :slightly_smiling_face: 
```suggestion
            if line.is_delivery:
                total_delivery += line.price_total
                continue
```